integration/utils/utils.go

```diff
@@ -0,0 +1,217 @@
+	c.Namespace = &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: giantswarmNamespace,
+		},
+	}
+
+	crdStorage, err := crdstorage.New(c)
+
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	l.Log("info", "booting crdstorage")
+	err = crdStorage.Boot(context.Background())
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return crdStorage, nil
+}
+
+func InitRangePool(crdStorage microstorage.Storage, l micrologger.Logger) (*rangepool.Service, error) {
+
+	rangePoolConfig := rangepool.DefaultConfig()
+	rangePoolConfig.Logger = l
+	rangePoolConfig.Storage = crdStorage
+
+	rangePool, err := rangepool.New(rangePoolConfig)
+	if err != nil {
+		return nil, microerror.Mask(err)
+
+	}
+
+	return rangePool, nil
+}
+
+func GenerateVNI(rangePool *rangepool.Service, clusterID string) (int, error) {
+	items, err := rangePool.Create(
+		context.Background(),
+		vniRangepoolNamespace,
+		rangePoolVNIID(clusterID),
+		1, // num
+		vniMin,
+		vniMax,
+	)
+
+	if err != nil {
+		return 0, microerror.Mask(err)
+	}
+
+	if len(items) != 1 {
+		return 0, microerror.Maskf(executionFailedError, "range pool VNI generation failed, expected 1 got %d", len(items))
+	}
+
+	return items[0], nil
+}
+
+func DeleteVNI(rangePool *rangepool.Service, clusterID string) error {
+	return rangePool.Delete(context.Background(), vniRangepoolNamespace, rangePoolVNIID(clusterID))
+}
+
+func GenerateIngressNodePorts(rangePool *rangepool.Service, clusterID string) (int, int, error) {
+	items, err := rangePool.Create(
+		context.Background(),
+		ingressRangepoolNamespace,
+		rangePoolIngressID(clusterID),
+		2, // num
+		nodePortMin,
+		nodePortMax,
+	)
+	if err != nil {
+		return 0, 0, microerror.Mask(err)
+	}
+
+	if len(items) != 2 {
+		return 0, 0, microerror.Maskf(executionFailedError, "range pool ingress port generation failed, expected 2 got %d", len(items))
+	}
+
+	return items[0], items[1], nil
+}
+
+func DeleteIngressNodePorts(rangePool *rangepool.Service, clusterID string) error {
+	return rangePool.Delete(context.Background(), ingressRangepoolNamespace, rangePoolIngressID(clusterID))
+}
+
+func GenerateFlannelNetwork(clusterID string, crdStorage microstorage.Storage, l micrologger.Logger) (string, error) {
+	var err error
+	var ipamConfig ipam.Config
+	{
+		ipamConfig.Logger = l
+		ipamConfig.Storage = crdStorage
+
+		var network *net.IPNet
+		_, network, err = net.ParseCIDR(flannelE2eNetworkRange)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+		ipamConfig.Network = network
+	}
+	ipamService, err := ipam.New(ipamConfig)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	cidrMask := net.CIDRMask(flannelCidrSize, 32)
+
+	cidr, err := ipamService.CreateSubnet(context.Background(), cidrMask, flannelNetworkAnnotation(clusterID))
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	return cidr.String(), nil
+}
+
+func DeleteFlannelNetwork(network string, crdStorage microstorage.Storage, l micrologger.Logger) error {
```

Comment:
For @giantswarm/sig-operator all this looks like the `kvm-operator` is less self contained than the other provider operators. Something to consider in the long run that we should try to improve the design and make responsibilities more clear. Right now they are quite scattered.